src/dlr_tensorflow/dlr_tensorflow.cc

```diff
@@ -70,73 +70,62 @@ void TensorflowModel::LoadFrozenModel(const char* pb_file) {
   TF_DeleteBuffer(graph_def);
 }
 
-void TensorflowModel::GenTensorSpec(bool is_input, const int batch_size) {
-  std::vector<std::string> tensor_names =
-      is_input ? input_names_ : output_names_;
+TF_Output TensorflowModel::ParseTensorName(const std::string& t_name) {
   std::regex r("^(.+):(\\d+)$");
-  for (std::string t_name : tensor_names) {
-    std::smatch match;
-    std::string op_name;
-    int op_out_id;
-    if (std::regex_search(t_name, match, r)) {
-      op_name = match.str(1);
-      op_out_id = std::stoi(match.str(2));
-    } else {
-      LOG(FATAL) << "ERROR: failed to parse tensor name " << t_name;
-      return;  // unreachable
-    }
+  std::smatch match;
+  if (!std::regex_search(t_name, match, r)) {
+    LOG(FATAL) << "ERROR: failed to parse tensor name " << t_name;
+  }
+  std::string op_name = match.str(1);
+  int op_out_id = std::stoi(match.str(2));
+  TF_Operation* op = TF_GraphOperationByName(graph_, op_name.c_str());
+  if (op == NULL) {
+    LOG(FATAL) << "ERROR: TF_GraphOperationByName failed for operation "
+               << op_name;
+  }
+  TF_Output oper_out = {op, op_out_id};
+  return oper_out;
+}
 
-    TF_Operation* op = TF_GraphOperationByName(graph_, op_name.c_str());
-    if (op == NULL) {
-      LOG(FATAL)
-          << "ERROR: inputOp TF_GraphOperationByName failed for operation "
-          << op_name;
-      return;  // unreachable
-    }
-    TF_Output oper_out = {op, op_out_id};
-    const int n_dim = TF_GraphGetTensorNumDims(graph_, oper_out, status_);
-    if (TF_GetCode(status_) != TF_OK) {
-      LOG(FATAL) << "ERROR: TF_GraphGetTensorNumDims failed "
-                 << TF_Message(status_);
-      return;  // unreachable
-    }
-    int64_t dims[n_dim];
-    TF_GraphGetTensorShape(graph_, oper_out, dims, n_dim, status_);
-    if (TF_GetCode(status_) != TF_OK) {
-      LOG(FATAL) << "ERROR: TF_GraphGetTensorShape failed "
-                 << TF_Message(status_);
-      return;  // unreachable
-    }
-    // Set fixed batch size if batch size is dynamic
-    if (dims[0] == -1) {
-      dims[0] = batch_size > 0 ? batch_size : 1;
-      TF_GraphSetTensorShape(graph_, oper_out, dims, n_dim, status_);
-      if (TF_GetCode(status_) != TF_OK) {
-        LOG(FATAL) << "ERROR: TF_GraphSetTensorShape failed "
-                   << TF_Message(status_);
-        return;  // unreachable
-      }
-    }
+void TensorflowModel::PrepInputs() {
+  for (int i = 0; i < num_inputs_; i++) {
+    const std::string t_name = input_names_[i];
+    const TF_Output oper_out = ParseTensorName(t_name);
+    const std::vector<int64_t> shape = input_shapes_[i];
```

Comment:
Make shape a reference so it doesn't have to be copied

Reply:
ok, fixed it in 3 places
```
diff --git a/src/dlr_tensorflow/dlr_tensorflow.cc b/src/dlr_tensorflow/dlr_tensorflow.cc
index b1bcb64..bae473e 100644
--- a/src/dlr_tensorflow/dlr_tensorflow.cc
+++ b/src/dlr_tensorflow/dlr_tensorflow.cc
@@ -89,9 +89,9 @@ TF_Output TensorflowModel::ParseTensorName(const std::string& t_name) {
 
 void TensorflowModel::PrepInputs() {
   for (int i = 0; i < num_inputs_; i++) {
-    const std::string t_name = input_names_[i];
+    const std::string& t_name = input_names_[i];
     const TF_Output oper_out = ParseTensorName(t_name);
-    const std::vector<int64_t> shape = input_shapes_[i];
+    const std::vector<int64_t>& shape = input_shapes_[i];
     const int64_t* dims = shape.data();
     const int n_dim = shape.size();
     size_t num_elements = 1;
@@ -120,7 +120,7 @@ void TensorflowModel::PrepInputs() {
 }
 
 void TensorflowModel::PrepOutputs() {
-  for (std::string t_name : output_names_) {
+  for (std::string& t_name : output_names_) {
     TF_Output oper_out = ParseTensorName(t_name);
 
     outputs_.push_back(oper_out);
```